File: nexabank/frontend/src/app/core/services/transaction.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Transaction, TransferRequest } from '../../shared/models/transaction.model';

export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
}

@Injectable({ providedIn: 'root' })
export class TransactionService {
  private http = inject(HttpClient);

  transfer(request: TransferRequest): Observable<ApiResponse<Transaction>> {
    return this.http.post<ApiResponse<Transaction>>(
      `${environment.apiUrl}/api/transactions/transfer`,
      request
    );
  }

  getTransactionsByAccount(accountId: number): Observable<ApiResponse<Transaction[]>> {
    return this.http.get<ApiResponse<Transaction[]>>(
      `${environment.apiUrl}/api/transactions/account/${accountId}`
    );
  }

  getTransaction(id: number): Observable<ApiResponse<Transaction>> {
    return this.http.get<ApiResponse<Transaction>>(
      `${environment.apiUrl}/api/transactions/${id}`
    );
  }
}

File: nexabank/transaction-service/src/main/java/com/nexabank/transaction/service/TransactionService.java
package com.nexabank.transaction.service;

import com.nexabank.transaction.dto.ApiResponse;
import com.nexabank.transaction.dto.TransactionResponse;
import com.nexabank.transaction.dto.TransferRequest;

import java.util.List;

/**
 * Transaction service interface.
 */
public interface TransactionService {

    TransactionResponse transfer(TransferRequest request, String userId);

    List<TransactionResponse> getTransactionsByAccountId(Long accountId);

    TransactionResponse getTransactionById(Long id);
}
